amlb/datasets/file.py

```diff
@@ -33,32 +34,25 @@ def __init__(self, cache_dir=None):
     def load(self, dataset, fold=0):
         dataset = dataset if isinstance(dataset, ns) else ns(path=dataset)
         log.debug("Loading dataset %s", dataset)
+        target = dataset['target']
+        type_ = dataset['type']
+        features = dataset['features']
+
+        if DatasetType[type_] == DatasetType.timeseries:
```

Comment:
```suggestion
        if type_ and DatasetType[type_] == DatasetType.timeseries:
```
`type_` may be `None` which gives a `KeyError` which made the unit tests fail.
Could you also add a unit test in `tests/unit/amlb/datasets/file` that loads a timeseries dataset and some tests that assert the validation logic?

Reply:
Thanks! Done

Reply:
Sorry for causing so many "round trips" :/

Reply:
@PGijsbers I've added tests and force-added a small dataset sample to `tests/unit/amlb/datasets/file/resources/m4_hourly_subset.csv`. Do you think there are any other scenarios we should cover with tests?

Reply:
I think we should at least cover the scenario where the dataset's `timestamp` and `id_column` are non-default raise the right errors if not specified, and do load when they are specified as well as test that the loader correctly determines when a dataset has insufficient samples to evaluate the required amount of folds.

Reply:
@PGijsbers Done 👍